src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java

```diff
@@ -420,17 +483,24 @@ private String handleNotificationsCaSolicitorCreatedCase(CaseData caseData, Stri
                 whoIsResponsibleForServing = caseData.getApplicants().get(0).getValue().getRepresentativeFullName();
                 //This is added with assumption that, For applicant legl representative selection
                 // if multiple applicants are present only the first applicant solicitor will receive notification
-                List<Document> packHiDocs = getNotificationPack(caseData, PrlAppsConstants.HI, c100StaticDocs);
-                packHiDocs.addAll(c100StaticDocs);
+
+                List<Document> packHiDocs = new ArrayList<>();
+                caseData.getRespondents().forEach(respondent -> packHiDocs.add(generateAccessCodeLetter(authorization, caseData,
+                                                                                                        respondent,
+                                                                                                        null,
+                                                                                                        PRL_LET_ENG_C100_RE6))
+                );
+                packHiDocs.addAll(getNotificationPack(caseData, PrlAppsConstants.HI, c100StaticDocs));
                 Map<String, Object> dynamicData = EmailUtils.getCommonSendgridDynamicTemplateData(caseData);
                 dynamicData.put("name", caseData.getApplicants().get(0).getValue().getRepresentativeFullName());
+                dynamicData.put("c100", true);
```

Comment:
why is this field?

Reply:
for c100 and fl401 there is a slight change in text between templates, so using same template with if condition for text change